R/splatPop-simulate.R

```diff
@@ -0,0 +1,849 @@
+#' splatPop simulation
+#'
+#' Simulate scRNA-seq count data using the splat model for a population of
+#' individuals with correlation structure.
+#'
+#' @param vcf vcfR object containing genotypes of samples to simulate.
+#' @param params splatPopParams object containing parameters for population
+#'        scale simulations. See \code{\link{splatPopParams}} for details. 
+#'        Default=`newSplatPopParams()`.
+#' @param method which simulation method to use. Options are "single" which
+#'        produces a single population, "groups" which produces distinct groups
+#'        (eg. cell types), "paths" which selects cells from continuous
+#'        trajectories (eg. differentiation processes).
+#' @param gff Either FALSE (if random.genes=TRUE) or a data.frame object 
+#'        containing a GFF/GTF file.
+#' @param key Either FALSE or a data.frame object containing a full or partial
+#'        splatPop key.
```

Comment:
The key is one of the 3 objects returned by splatPopSimulate, it contains information about the genes simulated, including mean/cv, eQTL assignments, DE, etc. The reason it is also an optional input, is that you could provide splatPopSimulateMeans with the key and it would simulate a new population using the same parameters. Or if you had real genes you want to simulate SC expression for, you can provide the genes their location and their known bulk expression levels, and let splatPopSimulate fill in the rest. 

It is a bit clunky, but this feature was strongly requested by one of our collaborators.

Reply:
👍 That sounds like a reasonable feature, still think we can come up with a better name though 😼? I think we can have this kind of optional things as part of `splatPopParams` but depends how hard it would be to implement (and the size issue).

Reply:
I've kept the splatPop key object as an output of splatPopSimulateMeans, but now, when you run splatPopSimulate the information from that key is now stored in the sce object in the rowData (since they key contains info for each gene).